lib/internal/bootstrap/node.js

```diff
@@ -102,6 +102,31 @@ process.domain = null;
 }
 process._exiting = false;
 
+{
+  const warnIntegerCoercionDeprecation = deprecate(
+    () => {},
+    'Implicit coercion to integer for exit code is deprecated.',
+    'DEP0164'
+  );
+
+  let exitCode;
+
+  ObjectDefineProperty(process, 'exitCode', {
+    __proto__: null,
+    get() {
+      return exitCode;
+    },
+    set(code) {
+      if (perThreadSetup.isDeprecatedExitCode(code)) {
+        warnIntegerCoercionDeprecation();
+      }
+      exitCode = code;
+    },
+    enumerable: true,
+    configurable: false,
```

Comment:
i believe this line caused #45683. was this intentional? `delete process.exitCode` has always worked in the past.

Reply:
This also makes it impossible to spy on `process.exitCode` in tests, making it more difficult to test libraries that set `exitCode`.

Reply:
cc @daeyeon @aduh95

Reply:
Not sure it was intentional, I would suggest opening a PR turning it to `configurable: true` and see if anyone objects.

Reply:
Thanks; filed #49579